include/zephyr/net/wifi_utils.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Utility functions to be used by the Wi-Fi subsytem.
```

Comment:
This is not expected to be a separate module, just an extension to wifi_mgmt. Added this to the `wifi_mgmt` group.

Reply:
Right, but I guess where I was coming from was more along the lines of making the documentation for wifi_mgmt less "flat" and crowded (see https://builds.zephyrproject.io/zephyr/pr/60686/docs/doxygen/html/group__wifi__mgmt.html), and some grouping wouldn't hurt (text helpers, parser utils, ...). Right now, this "flatness" may IMO unnecessarily give an impression of Wifi_mgmt being a "catch all" management module (which I guess it is tho :)) and it would be great to help users navigate it more easily.
You could consider using `@name { ... }` sections if you're not comfortable with a group showing up as a "Module".

Reply:
> Right now, this "flatness" may IMO unnecessarily give an impression of Wifi_mgmt being a "catch all" management module (which I guess it is tho :)) and it would be great to help users navigate it more easily.
> You could consider using `@name { ... }` sections if you're not comfortable with a group showing up as a "Module".

In this specific case `wifi_utils` is core part of `wifi_mgmt`, so, even though we add a new group or name, it will be shown along with `wifi_mgmt` docs, so, IMHO, no need for a separate group.

Reply:
@krish2718 While I am not questioning it's a "core part" of wifi_mgmt, wouldn't something along the lines of the following help bring more structure to the reader? (BTW, shouldn't there be a consistent way to get human-readable strings, vs. the current mix of helper functions and string arrays?)

<img width="835" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/128251/c3f8bbb6-56e8-4554-9649-945a8f8e19f6">

Reply:
Sure, it's definitely good to have. But so far in Zephyr, we had let doxygen create standard `C` groups (functions, macros, data structures etc), haven't done any manual grouping, so, requires a wider overhaul than this PR.

Reply:
> Sure, it's definitely good to have. But so far in Zephyr, we had let doxygen create standard `C` groups (functions, macros, data structures etc), haven't done any manual grouping, so, requires a wider overhaul than this PR.

I would disagree, there are many areas where `@name`d sections are being defined and they really help bringing clarity. See e.g https://docs.zephyrproject.org/latest/doxygen/html/group__gpio__interface.html.
You are right that it would be great to have a larger Zephyr-wide overhaul (I am actually doing a lot of Doxygen cleanup these days), but I don't see at as a valid reason for not adding grouping wherever it makes sense. Plus, in this particular instance, I think that doing the "grouping exercise" is also good to help surfacing inconsistencies in the API w.r.t how to get human-readable string for a given enum.

Reply:
Thanks for the link. Using categories is definitely helpful but all I am saying is we can take this up in another doc improvement PR. Of course we can add `wifi_utils` as a separate `@name` in this PR itself.

Reply:
Thanks both. I have added a separate @name section for wifi_utils.